atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableTimestampsReader.java

```diff
@@ -0,0 +1,98 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RangeRequests;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.schema.generated.SweepableTimestampsTable;
+import com.palantir.atlasdb.schema.generated.TargetedSweepTableFactory;
+import com.palantir.atlasdb.sweep.Sweeper;
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.util.PersistableBoolean;
+
+public class SweepableTimestampsReader {
+    private static final TableReference TABLE_REF = TargetedSweepTableFactory.of()
+            .getSweepableTimestampsTable(null).getTableRef();
+    private final KeyValueService kvs;
+    private final SweepTimestampProvider timestampProvider;
+    private final KvsSweepQueueProgress progress;
+
+    SweepableTimestampsReader(KeyValueService kvs, SweepTimestampProvider provider) {
+        this.kvs = kvs;
+        this.timestampProvider = provider;
+        this.progress = new KvsSweepQueueProgress(kvs);
+    }
+
+    public Optional<Long> nextSweepableTimestampPartition(int shard, boolean conservative) {
+        long lastSweptPartitionFine = progress.getLastSweptTimestampPartition(shard, conservative);
+        long lastSweptPartitionCoarse = SweepQueueUtils.partitionFineToCoarse(lastSweptPartitionFine);
+        return nextSweepablePartition(shard, conservative, lastSweptPartitionCoarse, lastSweptPartitionFine);
+    }
+
+    private Optional<Long> nextSweepablePartition(int shard, boolean conservative, long sweptCoarse, long sweptFine) {
+        long sweepTimestamp = timestampProvider.getSweepTimestamp(Sweeper.fromBoolean(conservative));
+        long sweepTimestampFine = SweepQueueUtils.tsPartitionFine(sweepTimestamp);
+        long sweepTimestampCoarse = SweepQueueUtils.tsPartitionCoarse(sweepTimestamp);
+        long currentCoarse = sweptCoarse;
+
+        while (currentCoarse <= sweepTimestampCoarse) {
+            Optional<Long> candidateFine = getCandidatesInCoarsePartition(shard, currentCoarse, conservative).stream()
+                    .filter(ts -> ts > sweptFine && ts < sweepTimestampFine)
+                    .min(Long::compareTo);
+            if (candidateFine.isPresent()) {
+                return candidateFine;
+            }
+            currentCoarse++;
+        }
+        return Optional.empty();
```

Comment:
Do we actually need to do this on the client side? I'm thinking we can do a `getRange` or `getRowsColumnRange` with lower bound `(sweptCoarse, sweptFine)` and upper bound `(sweepTsCoarse, sweepTsFine)`